astropy/units/core.py

```diff
@@ -2552,7 +2552,7 @@ def _condition_arg(value):
     ValueError
         If value is not as expected
     """
-    if isinstance(value, (np.ndarray, float, int, complex)):
+    if isinstance(value, (np.ndarray, float, int, complex, np.void)):
```

Comment:
Looking at this, I think the ``float, int, complex`` can be compared to just ``numbers.Number``.
And this will miss ``np.float64(10.0)``, which is ``np.generic``.

The list might be more complete as ``(np.ndarray, Number, np.generic, np.void)``.

Reply:
But ``np.void`` fixes this bug!

Reply:
@nstarman - I never liked this whole `_condition_arg` much, so my tendency would be to merge the fix here and then see if we cannot get rid of it altogether...

Reply:
Even better. Then `ndarray`-like might work